v2/apiserver/internal/core/projects.go

```diff
@@ -200,6 +206,71 @@ func (p *projectsService) Create(
 			errors.Wrapf(err, "error storing new project %q", project.ID)
 	}
 
+	// Make the current user an admin, developer, and user of the project
+	principal := libAuthn.PrincipalFromContext(ctx)
+
+	var principalRef authz.PrincipalReference
+	switch prin := principal.(type) {
+	case *authn.User:
+		principalRef = authz.PrincipalReference{
+			Type: authz.PrincipalTypeUser,
+			ID:   prin.ID,
+		}
+	case *authn.ServiceAccount:
+		principalRef = authz.PrincipalReference{
+			Type: authz.PrincipalTypeServiceAccount,
+			ID:   prin.ID,
+		}
+	default:
+		return project, nil
+	}
+
+	if err = p.roleAssignmentsStore.Grant(
```

Comment:
I was thinking perhaps simply granting admin rights would be sufficient (it'd already come with authorization that the developer and user role assignments would bring.)  Can you elaborate on the need for explicitly adding all?

Reply:
> it'd already come with authorization that the developer and user role assignments would bring

Partially accurate.

As things stand currently, an admin can manage permissions, secrets, and delete the project.

A user can _use_ the project (like create events), but cannot modify it in any way.

A developer can update the definition of the project.

These are deliberately separate. You can easily imagine three different personas fulfilling these roles.

Imagine a team lead is the admin since he or she might have access to secrets that others are not privy to.

Imagine that a member of the team is a developer.

Imagine that some person in a business unit is a user.

Now..... all of that said, someone who is an admin _can_ grant themselves the other two roles, but they don't have them automatically.

My intent here was that when a project is created, the party who created it is automatically given the _full_ range of things that can be done with/to the project.

Happy to discuss more if you think this can be improved upon.

Reply:
Thanks for laying out those scenarios; definitely makes sense.

What still hasn't explicitly been mentioned is that a project admin (or, strictly speaking, a principal whose only roleAssigment for a project is RoleProjectAdmin) is not automatically authorized to use or develop on the project itself (at least in regards to the authorizations that the associated roleAssignments for those two bring), right?  If I have this right, I think it is okay for now... it may just be somewhat unexpected for some Brigade users ('users' in a general sense here).  Or, it may not.  Therefore, it seems prudent to go with what you've laid out until (if) we get feedback in this area.

Reply:
The help says this:

> ADMIN      Grant the ADMIN project role, which enables management of all apects of the project, including its secrets, as well as project-level permissions for other users and service accounts.

Similar comments exist in the code.

You're right that "enables management of all apects of the project" (there's a typo in there, isn't there?) could be misleading.